src/App.ts

```diff
@@ -524,6 +527,23 @@ export default class App {
     return this.receiver.stop(...args);
   }
 
+  /**
+   * Register subcription middleware
+   *
+   * @param listeners middleware that process and react to subscription payloads
+   */
+  public subscription(options: SubscriptionOptions): this {
```

Comment:
Note: Originally the design called for this method to accept a first parameter for a `custom_id` that would correspond to specific resource domains (set up during Slack App configuration). The `custom_id` would have had to be sent from Slack backend. Since this feature will not be implemented at this time, adding this custom_id field later would be a breaking change.